spring-integration-file/src/main/java/org/springframework/integration/file/remote/synchronizer/AbstractInboundFileSynchronizer.java

```diff
@@ -233,6 +233,12 @@ public void synchronizeToLocalDirectory(final File localDirectory) {
 
 	@Override
 	public void synchronizeToLocalDirectory(final File localDirectory, final int maxFetchSize) {
+		if (maxFetchSize == 0) {
+			if (logger.isDebugEnabled()) {
+				logger.debug("Max Fetch Size is zero - fetch ignored");
```

Comment:
OK. But without extra `isDebugEnabled()`.
Although this is a good optimization already anyway for the entire `synchronizeToLocalDirectory()` ?